types/src/v0/impls/block/full_payload/payload.rs

```diff
@@ -68,13 +68,23 @@ impl Payload {
         // add each tx to its namespace
         let mut ns_builders = BTreeMap::<NamespaceId, NsPayloadBuilder>::new();
         for tx in transactions.into_iter() {
-            // accounting for block byte length limit
-            block_byte_len += tx.payload().len() + NsPayloadBuilder::tx_table_entry_byte_len();
+            let mut tx_size = tx.payload().len() + NsPayloadBuilder::tx_table_entry_byte_len();
             if !ns_builders.contains_key(&tx.namespace()) {
                 // each new namespace adds overhead
-                block_byte_len +=
+                tx_size +=
                     NsTableBuilder::entry_byte_len() + NsPayloadBuilder::tx_table_header_byte_len();
             }
+
+            if tx_size > max_block_byte_len {
+                // skip this transaction since it excceds the block size limit
+                tracing::warn!(
+                    "skip the transaction to fit in maximum block byte length {max_block_byte_len}, transaction size {tx_size}"
+                );
+                continue;
+            }
+
+            // accounting for block byte length limit
+            block_byte_len += tx_size;
             if block_byte_len > max_block_byte_len {
                 tracing::warn!("transactions truncated to fit in maximum block byte length {max_block_byte_len}");
```

Comment:
This isn't part of this PR but if we have a large transaction (that is not larger than the block) we skip adding any transactions behind it even if more transactions may fit into the block.

Reply:
I guess because we cannot change the order of the transactions, so I just keep this logic.